ingestion/src/metadata/ingestion/source/database/dbt/metadata.py

```diff
@@ -629,154 +637,171 @@ def create_dbt_tests_suite_definition(
-                test_case_status = TestCaseStatus.Failed
+            manifest_node = dbt_test.get("manifest_node")
+            if manifest_node:
+                logger.info(
+                    f"Processing DBT Test Case Results for node: {manifest_node.name}"
+                )
+                dbt_test_result = dbt_test.get("results")
+                test_case_status = TestCaseStatus.Aborted
                 test_result_value = 0
+                if dbt_test_result.status.value in {"success", "pass"}:
+                    test_case_status = TestCaseStatus.Success
+                    test_result_value = 1
+                elif dbt_test_result.status.value in {"failure", "fail"}:
+                    test_case_status = TestCaseStatus.Failed
+                    test_result_value = 0
+
+                # Process the Test Timings
+                dbt_test_timings = dbt_test_result.timing
+                dbt_test_completed_at = None
+                for dbt_test_timing in dbt_test_timings:
+                    if dbt_test_timing.name == "execute":
+                        dbt_test_completed_at = dbt_test_timing.completed_at
+                dbt_timestamp = None
+                if dbt_test_completed_at:
+                    dbt_timestamp = dbt_test_completed_at.timestamp()
+
+                # Create the test case result object
+                test_case_result = TestCaseResult(
+                    timestamp=dbt_timestamp,
+                    testCaseStatus=test_case_status,
+                    testResultValue=[
+                        TestResultValue(
+                            name=dbt_test_result.unique_id,
+                            value=str(test_result_value),
+                        )
+                    ],
+                )
 
-            # Process the Test Timings
-            dbt_test_timings = dbt_test_result["timing"]
-            dbt_test_completed_at = None
-            for dbt_test_timing in dbt_test_timings:
-                if dbt_test_timing.get("name", "") == "execute":
-                    dbt_test_completed_at = dbt_test_timing.get("completed_at")
-            dbt_timestamp = None
-            if dbt_test_completed_at:
-                dbt_timestamp = datetime.strptime(
-                    dbt_test_completed_at, "%Y-%m-%dT%H:%M:%S.%fZ"
-                ).replace(microsecond=0)
-                dbt_timestamp = dbt_timestamp.timestamp()
-
-            # Create the test case result object
-            test_case_result = TestCaseResult(
-                timestamp=dbt_timestamp,
-                testCaseStatus=test_case_status,
-                testResultValue=[
-                    TestResultValue(
-                        name=dbt_test_result.get("unique_id"),
-                        value=str(test_result_value),
+                # Create the test case fqns and add the results
+                for table_fqn in dbt_test.get("upstream"):
+                    source_elements = table_fqn.split(".")
```

Comment:
We have a constant named `FQN_SEPARATOR` for this. Let's import it from https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/utils/fqn.py#L48